core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckServiceBuilder.java

```diff
@@ -105,6 +119,44 @@ public HealthCheckServiceBuilder healthyResponse(AggregatedHttpResponse healthyR
         return this;
     }
 
+    /**
+     * Sets the {@link AggregatedHttpResponse} to send when the {@link Service} is degraded. The following
+     * response is sent by default:
+     *
+     * <pre>{@code
+     * HTTP/1.1 200 OK
+     * Content-Type: application/json; charset=utf-8
+     *
+     * { "healthy": true, "status": "DEGRADED" }
+     * }</pre>
+     *
+     * @return {@code this}
+     */
+    public HealthCheckServiceBuilder degradedResponse(AggregatedHttpResponse degradedResponse) {
```

Comment:
note: from the perspective of health checking in envoy, just sending a `x-envoy-degraded` header to denote an endpoint is probably enough. Checked that this won't interfere with ongoing xDS related changes.

Reply:
It seems envoy use headers to denote the specific status. What would be more appropriate for armeria to denote specific status? The headers or status field in the json?

Reply:
I think it's fine to let users customize the response for now - later depending on requirements we may just introduce a separate `EnvoyHealthCheckService`.

Reply:
We probably should do some research on the behavior of well known server implementations and make the default response covers them all, as a follow-up to this PR.